### ISSUE TITLE
Import files to track database (Engine DJ v3.3.0) Incorrect number of bindings supplied to SQL statement 47 not 48

### ISSUE THREAD

**Issue by Trickster138** [ssabug/piratengine#1]
ben@Bens-MacBook-Air-388 piratengine-master % /usr/local/bin/python3 /Users/ben/Music/piratengine-master/piratengine.py
2024-03-08 15:45:01.695588 [ PRTE ] ###### WELCOME TO PIRATENGINE
2024-03-08 15:45:01.695689 [ GUID ] Initialisation start
<class 'modules.gui.MainWindow'>
2024-03-08 15:45:03.699297 [ GUIM ] Load Database clicked
2024-03-08 15:45:03.711496 [ ENDB ] connecting to database /Users/ben/Music/Engine Library/Database2/m.db
2024-03-08 15:45:03.712072 [ ENDB ] db total_changes = 0
2024-03-08 15:45:03.712164 [ ENDB ] Reading database tables ...
2024-03-08 15:45:03.712238 [ ENDB ]   table : Playlist
2024-03-08 15:45:03.713133 [ ENDB ]   table : PlaylistEntity
2024-03-08 15:45:03.717138 [ ENDB ]   table : Track
2024-03-08 15:45:03.734091 [ ENDB ]   table : AlbumArt
2024-03-08 15:45:03.773974 [ ENDB ]   table : ChangeLog
2024-03-08 15:45:03.774308 [ ENDB ]   table : Information
2024-03-08 15:45:03.774409 [ ENDB ]   table : Pack
2024-03-08 15:45:03.774485 [ ENDB ]   table : PerformanceData
2024-03-08 15:45:03.777671 [ ENDB ]   table : PlayListAllChildren
2024-03-08 15:45:03.777873 [ ENDB ]   table : PlaylistAllParent
2024-03-08 15:45:03.777994 [ ENDB ]   table : PlaylistPath
2024-03-08 15:45:03.778510 [ ENDB ]   table : sqlite_sequence
2024-03-08 15:45:03.778632 [ ENDB ] Read 12 tables
2024-03-08 15:45:03.778673 [ ENDB ] database schema version : 2.21.0
2024-03-08 15:45:07.935976 [ ENDB ] Scanning files in directory : /Users/ben/Music/piratengine-master
2024-03-08 15:45:07.936366 [ ENDB ] Extensions : ['.mp3', '.wav', '.flac', '.ogg', '.MP3', '.WAV', '.FLAC', '.OGG']
2024-03-08 15:45:07.936510 [ ENDB ] Root folder : piratengine-master
2024-03-08 15:45:07.937426 [ ENDB ] Unscanned track found : ../Test Artist - Test Track.mp3
2024-03-08 15:45:07.938002 [ ENDB ] Number of added tracks : 1
2024-03-08 15:45:07.938070 [ ENDB ] Number of already present tracks : 0
2024-03-08 15:45:07.938137 [ ENDB ] Number of tracks in database : 1013
2024-03-08 15:45:10.902914 [ ENDB ] Adding /Users/ben/Music/Test Artist - Test Track.mp3 to Track database 
2024-03-08 15:45:10.903604 [ ENDB ] Relative path : ../ben/Music/Test Artist - Test Track.mp3
2024-03-08 15:45:10.954955 [ ENDB ] filename:Test Artist - Test Track.mp3
2024-03-08 15:45:10.955572 [ MAIN ] EXCEPTION ProgrammingError Incorrect number of bindings supplied. The current statement uses 48, and there are 47 supplied.
2024-03-08 15:45:10.956610 [ MAIN ] <FrameSummary file /Users/ben/Music/piratengine-master/modules/database.py, line 442 in insertVariableIntoTable>
2024-03-08 15:45:10.956676 [ MAIN ] <FrameSummary file /Users/ben/Music/piratengine-master/modules/gui.py, line 311 in ImportFilesButton_click>
2024-03-08 15:45:10.956731 [ MAIN ] <FrameSummary file /Users/ben/Music/piratengine-master/modules/gui.py, line 41 in __init__>
2024-03-08 15:45:10.956783 [ MAIN ] <FrameSummary file /Users/ben/Music/piratengine-master/piratengine.py, line 247 in initGui>
2024-03-08 15:45:10.956834 [ MAIN ] <FrameSummary file /Users/ben/Music/piratengine-master/piratengine.py, line 265 in <module>>
2024-03-08 15:45:10.956893 [ GUIM ] track added : /Users/ben/Music/Test Artist - Test Track.mp3

**Comment by ssabug** [comment]
what database version are you using?

**Comment by ssabug** [comment]
Ok so looks like there are different fields in track table in your version of the database.
Can you please provide me a sample file of Database/m.db or at least a list of all fields in the Track table (you can use DbBrowser for SQlite) ?

**Comment by ssabug** [comment]
I implemented Track table changes that i thought were added in version 3 for version > 2.20 , i updated the code, if you want to have a try.

Be careful to use **Scan folder for music files** feature only to scan the same folder as the one the database files are attached. For now, the program is only able to scan for present tracks on the device attached to the edited database.

In the future this feature will copy the file automatically if not already present in device/database, but ive forgot to notify this temporary situation.
